gcloud/translate/client.py

```diff
@@ -0,0 +1,199 @@
+                  ISO 639-1 language code (using the dictionary key
+                  ``language``). If ``target_language`` is passed, each
+                  dictionary will also contain the name of each supported
+                  language (localized to the target language).
+        """
+        query_params = {'key': self.key}
+        if target_language is not None:
+            query_params['target'] = target_language
+        response = self.connection.api_request(
+            method='GET', path='/languages', query_params=query_params)
+        return response.get('data', {}).get('languages', ())
+
+    def detect_language(self, *values):
+        """Detect the language of a string or list of strings.
+
+        See: https://cloud.google.com/translate/v2/\
+        detecting-language-with-rest
+
+        :type values: tuple
+        :param values: Tuple of strings that will have language detected.
+
+        :rtype: list
+        :returns: A list of dictionaries for each queried value. Each
+                  dictionary typically contains three keys (though not
+                  all will be present in all cases)
+
+                  * ``confidence``: The confidence in language detection, a
+                    float between 0 and 1.
+                  * ``input``: The corresponding input value.
+                  * ``language``: The detected language (as an ISO 639-1
+                    language code).
+
+                  If multiple languages are detected for a given input, then
+                  the there will be a list of dictionaries (instead of a single
+                  dictionary) for that queried value.
+        :raises: :class:`ValueError <exceptions.ValueError>` if the number of
+                 detections is not equal to the number of values.
+        """
+        query_params = [('key', self.key)]
+        query_params.extend(('q', _to_bytes(value, 'utf-8'))
+                            for value in values)
+        response = self.connection.api_request(
+            method='GET', path='/detect', query_params=query_params)
+        detections = response.get('data', {}).get('detections', ())
+
+        if len(values) != len(detections):
+            raise ValueError('Expected same number of values and detections',
+                             values, detections)
+
+        for index, value in enumerate(values):
+            for detection in detections[index]:
+                detection['input'] = value
+                # The ``isReliable`` field is deprecated.
+                detection.pop('isReliable', None)
+            # If there was only one detected match, replace the
+            # list of matches with the single match. Empirically, even
+            # clearly ambiguous text like "no" only returns a single
+            # detection.
+            if len(detections[index]) == 1:
+                detections[index] = detections[index][0]
+
+        return detections
+
+    def translate(self, *values, **kwargs):
```

Comment:
I had that in the original commit but changed it in the 2nd commit. I just wanted to save the users from typing `[` and `]`.

We could also do what we did in datastore and have `translate` and `translate_multiple` methods?

Also I just naturally added `_language` to source and target since as standalone words they are a bit vague. I can shorten them if you prefer JJ.

As for CIDs, it was trivial to add it as a kwarg, I can remove it but it isn't really hurting anyone.